django_rest_generator/client.py

```diff
@@ -227,46 +223,51 @@ def _build_resource_object(self, resource_name: str, endpoints: dict):
 
         resource_mixins = set()
         custom_actions = set()
-        for endpoint, methods in endpoints.items():
-            if endpoint == "/":
+        for endpoint in resource.endpoints:
+            if endpoint.path == "/":
                 # It is a root object or a singleton
-                for method in methods:
-                    resource_mixins.update(object_method_map[method])
-            elif re.match(r"{.*}\/$", endpoint) is not None:
+                for operation in endpoint.operations:
+                    resource_mixins.update(object_method_map[operation.method])
+            elif re.match(r"{.*}\/$", endpoint.path) is not None:
                 # It is an instance/object endpoint
-                for method in methods:
-                    resource_mixins.update(instance_method_map[method])
+                for operation in endpoint.operations:
+                    resource_mixins.update(instance_method_map[operation.method])
 
             else:
                 # We are a custom action here
                 # A custom action must only have one method.
-                if len(methods) > 1:
+                if len(endpoint.operations) > 1:
                     raise Warning(
-                        f"Custom action at {resource_name}/{endpoint} has more than the ONE allowed HTTP method: {methods}."
+                        f"Custom action at {resource.name}/{endpoint.path} has more than the ONE allowed HTTP method: {[ep.method for ep in endpoint.operations]}."
```

Comment:
we raise an error here and stop
i feel like it should be a more explicit error

Reply:
I could make a custom error for this case, something like `MoreThanOneEndpointException` but in reality this shouldn't block us from using the API altogether, only that resource wouldn't be available

Reply:
the warning just feels weird

Reply:
mmm, so what would you suggest considering that this should not be blocking? Perhaps adding a logger and just logging it without raising the warning? I feel like a warning a appropriate for it but I'm open to suggestions.